### PR TITLE
Location Puck

### PR DESCRIPTION
**Pull request by dbabbs**
#### Description

Add location puck components to Base web component library. 
[Design files [internal only]](https://www.figma.com/file/oO0NR8Uphi4gEjjBmLriU5/Map-modules-%5BWIP%5D?node-id=1381%3A67083)

#### API 
- `type`: choice of `consumer` or `earner`
- `confidenceRadius`: radius in px of the confidence circle. 
- `size`: only applies to earner puck. choice of `small`, `medium`, `large`
- `heading`: direction of the arrow. Accepts a number representing degrees between 0-360
- `showHeading`: renders the heading indicator. Only applies to the consumer location puck.

#### Scope
Minor: New Feature

image

### CODE REVIEW

File: src/locale/__tests__/unit.test.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ This source code is licensed under the MIT license found in the
 LICENSE file in the root directory of this source tree.
 */
 import React from 'react';
-import { render, getByTestId } from '@testing-library/react';
+import { render } from '@testing-library/react';
```
Comment (chasestarr) [inline src/locale/__tests__/unit.test.tsx:8]: Please delete this file. I renamed it in another PR

File: src/map-marker/location-puck.tsx [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+const ConsumerLocationPuckHeading = ({ heading }) => {
+  const [css, theme] = useStyletron();
+  return (
+    <svg
+      className={css({
+        height: `${6}px`,
+        width: `${11}px`,
+        position: 'absolute',
+        color: theme.colors.contentAccent,
+        transform: `rotate(${heading}deg) translateY(-16px)`,
```
Comment (chasestarr) [inline src/map-marker/location-puck.tsx:37]: `heading` is likely going to regularly change right? If so, let's move this css rule to `style` prop. Right now, styletron will create a new css class each time this value changes leading to a bunch of browser repaints from the new class added. 

```tsx
<svg
  style={{
    transform: `rotate(${heading}deg) translateY(-16px)`,
  }}
/>
```

File: src/map-marker/location-puck.tsx [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+};
+
+const EarnerLocationPuckHeading = ({ size, color, heading }) => {
+  const [css, theme] = useStyletron();
+  return (
+    <svg
+      className={css({
+        position: 'absolute',
+        transition: `${theme.animation.timing300} ${theme.animation.easeOutCurve} all`,
+        transform: `rotate(${heading}deg) scale(${EARNER_LOCATION_PUCK_CORE_SCALES[size]})`,
```
Comment (chasestarr) [inline src/map-marker/location-puck.tsx:96]: same suggestion to move this to `style` prop

File: src/map-marker/styled-components.js.flow [hunk 4]
```diff
@@ -5,11 +5,13 @@ This source code is licensed under the MIT license found in the
 LICENSE file in the root directory of this source tree.
 */
 // @flow
+import { number } from 'card-validator';
```
Comment (chasestarr) [inline src/map-marker/styled-components.js.flow:8]:
```suggestion
```

**Review by chasestarr — COMMENTED** [review]
Looks good overall, just a couple small edits

File: src/map-marker/styled-components.js.flow (hunk 4 above)
Comment (dbabbs) [reply src/map-marker/styled-components.js.flow:8]: ah! this must have snuck in from vscode autocomplete

**Review by chasestarr — APPROVED** [review]

**Pull request merged**